File: 200-number-of-islands/200-number-of-islands.cpp
class Solution {
public:
    int numIslands(vector<vector<char>>& grid) {
        int ans=0;
        int m=grid.size();
        int n=grid[0].size();
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(grid[i][j]=='1'){
                    vector<vector<bool>> visi(m,vector<bool>(n,false));
                    floodfill(grid,visi,i,j);
                    ans++;
                }
            }
        }
        
        return ans;
    }
    
    void floodfill(vector<vector<char>> &grid, vector<vector<bool>> &vis, int i, int j){
        // cout<<i<<","<<j<<endl;
        if (i<0 || i>=grid.size() || j<0 || j>=grid[0].size() || vis[i][j] || grid[i][j]=='0')
            return;
        vis[i][j]=true;
        grid[i][j]='0';
        floodfill(grid,vis,i+1,j);
        floodfill(grid,vis,i-1,j);
        floodfill(grid,vis,i,j+1);
        floodfill(grid,vis,i,j-1);
    }
};

File: 200-number-of-islands/200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        
        ans = 0
        
        def dfs(i,j):
            
            if i<0 or j<0 or i>=len(grid) or j>=len(grid[0]) or grid[i][j]=="0":
                return 
            
            grid[i][j]="0"
            
            dirs = [[0,1],[1,0],[-1,0],[0,-1]]
            for di,dj in dirs:
                dfs(i+di, j+dj)
                
            
            
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]=="1":
                    ans+=1
                    dfs(i,j)
                    
        return ans